custom_connectors/oauth2/zoho_mail_connector.rb

```diff
@@ -0,0 +1,540 @@
+                  { name: 'toAddress' },
+                  { name: 'content' },
+                  { name: 'ccAddress' },
+                  { name: 'bccAddress' }
+                ]
+              end
+    }
+
+  },
+
+  methods: {
+    get_account_id: lambda do |_connection|
+                      (get('/api/accounts')['data']
+                        .select { |a| a['type'] == 'ZOHO_ACCOUNT' })[0]
+                        .fetch('accountId')
+                    end,
+    get_from_addresses: lambda do |parsed|
+                          (parsed.select { |a| a['type'] == 'ZOHO_ACCOUNT' }
+                          )[0].fetch('sendMailDetails')
+                        end,
+
+    get_folders: lambda do |parsed|
+                   folder_json = []
+                   parsed.each do |folderdata|
+                     if folderdata['folderType'] == 'Inbox'
+                       folder_json << folderdata
+                     end
+                   end
+                   folder_json
+                 end,
+    add_content: lambda do |input|
+                   response = input['response']
+                   account_id = input['accountId']
+                   mail_content = []
+                   response.each do |mailjson|
+                     next if mailjson['folderId'].blank?
+
+                     content = get("/api/accounts/#{account_id}/folders"\
+                       "/#{mailjson['folderId']}/messages/"\
+                       "#{mailjson['messageId']}/content")['data']['content']
+                     mailjson = mailjson.merge('content' => content)
+                     mail_content << mailjson
+                   end
+                   mail_content
+                 end
+  },
+
+  pick_lists: {
+    from_address: lambda do |_connection|
+      call(:get_from_addresses,
+           get('/api/accounts')['data'])
+        .map do |from_address|
+        [from_address['fromAddress'],
+         from_address['fromAddress']]
+      end
+    end,
+
+    accounts: lambda do |_connection|
+      get('/api/accounts')['data']
+        .map do |accountdetails|
+        [accountdetails['displayName'],
+         accountdetails['accountId']]
+      end
+    end,
+
+    folders: lambda do |connection|
+      account_id = call(:get_account_id, connection)
+      call(:get_folders, get('/api/accounts' \
+        "/#{account_id}/folders")['data'])
+        .map { |folder| [folder['folderName'], folder['folderName']] }
+    end
+  },
+
+  actions: {
+
+    send_mail: {
+      description: "Send <span class='provider'>New mail</span> in " \
+        "<span class='provider'>Zoho Mail</span>",
+      input_fields: lambda do
+        [
+          { name: 'Account', control_type: 'select', pick_list:
+            'accounts', optional: true,
+            hint: 'Select the account (Zoho account/ POP account)
+             from the list of accounts available in Zoho Mail.' },
+          { name: 'From', control_type: 'select',
```

Comment:
This field should be a toggle field as well
https://docs.workato.com/developing-connectors/sdk/toggle-fields.html

Reply:
A user can have more than one mail accounts, this field lets them select the 'from address' in case of multiple from addresses. If the user chooses not to select then we choose the default account for the authenticated user. https://www.zoho.com/mail/help/dynamic-from-address.html

Reply:
Same applies to from address also. It can be confusing for user to manually enter. I'd ideally like this field to be picked from the list I show. If suppose wrong fromAddress in mapped, it can result in failures.